openedx/features/student_certificates/views.py

```diff
@@ -195,6 +195,34 @@ def download_certificate_pdf(request, certificate_uuid):
 
 
 def verify_certificate(request, key):
+    """
+    Provides verify certificate page
+
+    Arguments:
+        request: The request object.
+        key: Verification key of certificate.
+
+    Returns:
+        The verify certificate response.
+
+    """
+
     certificate_verification_key_obj = get_object_or_404(CertificateVerificationKey, verification_key=key)
     certificate_uuid = certificate_verification_key_obj.generated_certificate.verify_uuid
-    return redirect(reverse('shared_achievements', kwargs={'certificate_uuid': certificate_uuid}))
+
+    try:
+        certificate = GeneratedCertificate.eligible_certificates.get(
```

Comment:
We already have generated certificate in above statement at line 211, can we use that instead re-querying for it?